### DISCUSSION TITLE
[New] Ne charger des données/du code que lorsqu'elles sont/il est nécessaire

### DISCUSSION THREAD

**Discussion by tbroyer** [cnumr/best-practices#158] [Bonnes pratiques] [answered]
cf. https://github.com/cnumr/best-practices/discussions/109#discussioncomment-1545497

> faire un preload d'une resource qui sera de toutes façons utilisée dans la page elle-même, c'est de l'optimisation de performance, ce n'est ni bon ni mauvais d'un point de vu de l'écoconception ; faire un preload d'une autre page, ou d'une resource qui sera utilisée sur "les autres pages", ou qui _pourrait_ être utilisée sur la page elle-même en fonction des actions de l'utilisateur, c'est plutôt une mauvaise pratique d'écoconception

Ça s'applique finalement autant à du `<link rel="preload">` (`preconnect`/`prefetch`/`prerender`) qu'à du chargement de données (_via_ un `fetch`/`XMLHttpRequest`) ou encore la Cache API ([`cache.add()`](https://developer.mozilla.org/en-US/docs/Web/API/Cache/add)/[`cache.addAll()`](https://developer.mozilla.org/en-US/docs/Web/API/Cache/addAll))

**Comment by rlemaire** [comment]
Est-ce que ça ne redit pas des choses qu'on a déjà par ailleurs ? 

Par exemple BP#009, BP#022, BP#040, etc.

Peut-être qu'une modification de BP#040 suffirait ?

**Comment by tbroyer** [comment]
Ça rejoint effectivement un peu la BP#9 (dont la plupart des propositions disposent de leur propre BP par ailleurs), mais ici l'idée des auteurs serait (a priori, sauf erreur de programmation) de préparer des choses qui amélioreront les _performances_ d'une prochaine navigation **si** elle a lieu (partant du principe que ce sera le cas, chacun espère "capter" les utilisateurs sur plus d'une page). Si le parcours a lieu, alors ces ressources seront utilisées donc il n'y a pas "d'augmentation" du nombre de requêtes, et du coup pas de réduction possible (donc finalement la BP#9 ne s'applique pas vraiment).
L'idée de cette BP serait de leur rappeler que **si** cette navigation n'est pas faite, alors ils auront gâché des ressources.

Je ne vois pas le rapport avec les BP#22 et BP#40 par contre :man_shrugging:

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé l'issue: https://github.com/cnumr/best-practices/issues/282

**Comment by Fred-artwai** [comment]
C'est une excellente proposition. D'ailleurs, il faudrait en faire une BP dédiée aux images en systématisant l'usage du lazyloading pour les images en dessous de la ligne de flottaison notamment grâce à l'usage de l'attribut HTML loading. On ne charge les images que si elles doivent s'afficher !

**Reply by tbroyer** [reply]
Ça existe déjà: https://github.com/cnumr/best-practices/blob/main/chapters/BP_037_fr.md :wink: (voir aussi #44)
